karma.conf.js

```diff
@@ -1,21 +1,22 @@
 process.env.MOZ_HEADLESS = 1;
 
-if (!process.env.TRAVIS){
+if (!process.env.TRAVIS) {
     process.env.CHROME_BIN = require('puppeteer').executablePath();
 }
 
-module.exports = function (config) {
+module.exports = function(config) {
     var configuration = {
         frameworks: [
             'mocha',
             'browserify'
         ],
         files: [
-            'node_modules/@babel/polyfill/dist/polyfill.js', // For async/await in tests
+            'packages/web3/dist/web3.min.js',
+            'node_modules/@babel/polyfill/dist/polyfill.js',
```

Comment:
What I have done:
I've checked the logs during testing ``karma-parcel`` and the polyfills were missing in the e2e tests itself. I've tested the ``web3.js.min`` manually in the Chrome and Firefox browser. I've included the minified file into the HTML header and used the web3.js lib without any errors or further dependencies added to it. 

I currently can't see any way to track the inclusion of polyfills in the min file with a e2e test. The only way we could have further proof is to create a rollup plugin or to use an existing possibility which checks the inclusion of ``@babel/runtime`` in the bundle during the build time of the ``web3.min.js`` file.

Reply:
Ok then, if the manual check is correct then that will have to be enough. It's just one blind spot, so ok.

Reply:
Actually I'm unresolving this because I've opened a dummy PR on my fork which implements your initial idea in #3155. It strongly suggests that the headless browsers are consuming the minified file correctly and that the initial diagnosis of a missing built option / masking test config is correct. See here:

https://github.com/cgewecke/web3.js/pull/4

The travis output indicating the min is being pulled in by the require [is here][1]
[1]: https://travis-ci.com/cgewecke/web3.js/jobs/251343793#L594-L597

Reply:
Thanks for doing this test. Yes, this was the first fix I've applied and I've started after to extend the build pipeline to have CJS, ESM, and minified UMD bundles with source maps for each package. Browserify had some problems with processing all packages (it ran out of memory all the time). This was the reason why I overtook the configuration from 2.x and extended it. You are able to remove the polyfill injection because Browserify (at least as it looks like) does have those polyfills accessible globally and the new minified file doesn’t have that. Because we do have ``@babel/runtime`` listed as a dependency for each package and the transform runtime plugin in the plugin list are the required polyfills included in all minified files.

Reply:
I hope you don't mind, lets keep this unresolved for the moment until we are certain about this.

> You are able to remove the polyfill injection in your PR because Browserify (at least as it looks like) does have those polyfills accessible globally and the new minified file doesn’t have that. 

I'm not convinced by this argument. What we see in the *current* build is that unless the polyfill is injected into the Karma env, the tests fail. This is *not* because the browser env can't run the tests - it's because the bundle is corrupted (see #3155). In the example build I linked to above, your patch with 'useBuiltIns' fixes the bundle and allows the tests to run without injecting the polyfill.  

Why do you not think the same problem is possible here - e.g. something is missing from the bundling options that would make it possible for a clean run without adding a lib that (potentially) masks the problem?

Reply:
I ran `test:e2e:browsers` with this branch locally after removing `node_modules/@babel/polyfill/dist/polyfill.js` from the Karma files list (as in the other experiment) and everything worked.

Is it possible we are both correct here? The build is fine & the polyfill can be removed?

Reply:
Just tested it as well and yes it works now without including of the polyfill. The polyfill was required because it is bundling the non-minified version as well but isn't using it for the tests. The ``regeneratorRuntime`` error was then thrown on the loading of the not-used but bundled web3.js bundle. This error is no longer thrown because I've updated the ``babel`` config to transform the runtime for the ``cjs`` and ``esm`` bundles as well. This does guarantee us compatibility for older nodejs runtimes. Those two bundle types do have nodejs version 8 as target defined now.

Reply:
Ah! Makes sense. Thanks for explaining this to me.